btb/tuning/hyperparams/base.py

```diff
@@ -128,14 +138,35 @@ def transform(self, values):
         Returns:
             normalized (ArrayLike):
                 2D array of shape(len(values), self.K)
+
+        Examples:
+            >>> from btb.hyperparams.numerical import IntHyperParam
+            >>> ihp = IntHyperParam(min=1, max=4)
+            >>> ihp.transform(1)
+            array([[0.125]])
+
+            >>> ihp.transform([1, 2])
+            array([[0.125],
+                   [0.375]])
+
+            >>> ihp.transform([[1], [2]])
+            array([[0.125],
+                   [0.375]])
         """
-        if not isinstance(values, (list, np.ndarray)):
-            values = [values]
 
-        values = [[value] if not isinstance(value, (list, np.ndarray)) else value
-                  for value in values]
+        if not isinstance(values, np.ndarray):
+            values = np.asarray(values)
+
+        dimensions = len(values.shape)
+        if dimensions > 2:
+            raise ValueError('Too many dimensions.')
+
+        elif dimensions < 2:
+            values = values.reshape(-1, 1)
+
+        if values.shape[1] > 1:
+            raise ValueError('More than one columns are not supported.')
```

Comment:
This sentence should be improved: "Only one column is supported", o "Number of columns can only be one".